File: Reference/VB.NET/DynamicGeometry/Interaction/Actions/ActionCenter/History/IActionHistory.vb
Namespace Actions

	Friend Interface IActionHistory

		Sub RecordAction(ByVal NewAction As IAction)
		Sub Clear()

		Sub MoveBack()
		Sub MoveForward()

		Function CanMoveBack() As Boolean
		Function CanMoveForward() As Boolean
		Property Length() As Integer

	End Interface

End Namespace

File: Reference/C#/DynamicGeometry/Actions/History/IActionHistory.cs
using System.Collections.Generic;
using System.Collections.Specialized;

namespace GuiLabs.Utils.Actions
{
	internal interface IActionHistory : IEnumerable<IAction>, INotifyCollectionChanged
	{
		/// <summary>
		/// Appends an action to the end of the Undo buffer.
		/// </summary>
		/// <param name="newAction">An action to append.</param>
		/// <returns>false if merged with previous, else true</returns>
		bool AppendAction(IAction newAction);
		void Clear();

		void MoveBack();
		void MoveForward();

		bool CanMoveBack { get; }
		bool CanMoveForward { get; }
		int Length { get;}

		SimpleHistoryNode CurrentState { get; }

		IEnumerable<IAction> EnumUndoableActions();
	}
}
